crashtracker/src/collector/emitters.rs

```diff
@@ -94,6 +95,89 @@ pub unsafe fn emit_backtrace_by_frames(
     Ok(())
 }
 
+pub(crate) fn emit_crashreport(
+    pipe: &mut impl Write,
+    config: &CrashtrackerConfiguration,
+    config_str: &str,
+    metadata_string: &str,
+    signum: i32,
+) -> anyhow::Result<()> {
+    emit_metadata(pipe, metadata_string)?;
+    emit_config(pipe, config_str)?;
+    emit_siginfo(pipe, signum)?;
+    emit_procinfo(pipe)?;
+    pipe.flush()?;
+    emit_counters(pipe)?;
+    pipe.flush()?;
+    emit_spans(pipe)?;
+    pipe.flush()?;
+    emit_traces(pipe)?;
+    pipe.flush()?;
```

Comment:
Why is it that the emission of some data is immediately followed by a flush operation sometimes, but not always?

Reply:
Good question, it probably should always.  For a future PR.

Reply:
https://github.com/DataDog/libdatadog/issues/546